--- "SimpleWebServer - Kopie/include/simple-web-server/client_http.hpp"	2018-08-04 09:55:20.383171000 +0200
+++ SimpleWebServer/include/simple-web-server/client_http.hpp	2022-01-09 12:21:28.713487000 +0100
@@ -120,7 +120,7 @@
           timer = nullptr;
           return;
         }
-        timer = std::unique_ptr<asio::steady_timer>(new asio::steady_timer(socket->get_io_service()));
+        timer = std::unique_ptr<asio::steady_timer>(new asio::steady_timer(socket->get_executor()));
         timer->expires_from_now(std::chrono::seconds(seconds));
         auto self = this->shared_from_this();
         timer->async_wait([self](const error_code &ec) {
